exporter/awsemfexporter/datapoint.go

```diff
@@ -273,25 +375,72 @@ func (dps exponentialHistogramDataPointSlice) CalculateDeltaDatapoints(idx int,
 		bucketBegin = -math.Pow(base, float64(index+1))
 		metricVal := (bucketBegin + bucketEnd) / 2
 		count := negativeBucketCounts.At(i)
-		if count > 0 {
-			arrayValues = append(arrayValues, metricVal)
-			arrayCounts = append(arrayCounts, float64(count))
+		if count > 0 && firstSplit.length < firstSplit.maxBuckets {
+			firstSplit.values = append(firstSplit.values, metricVal)
+			firstSplit.counts = append(firstSplit.counts, float64(count))
+			firstSplit.length++
+			currentLength++
+			if firstSplit.maxBuckets < totalBucketLen {
+				firstSplit.count += int(count)
+				if currentLength == firstSplit.length {
+					firstSplit.bucketMin = bucketEnd
+					if bucketBegin < bucketEnd {
+						firstSplit.bucketMin = bucketBegin
+					} else {
+						firstSplit.bucketMin = bucketEnd
+					}
+				}
+			}
+		} else if count > 0 {
+			if currentLength == firstSplit.maxBuckets && currentLength != 0 {
+				if bucketBegin < bucketEnd {
+					secondSplit.bucketMax = bucketEnd
+				} else {
+					secondSplit.bucketMax = bucketBegin
+				}
+			}
+			secondSplit.values = append(secondSplit.values, metricVal)
+			secondSplit.counts = append(secondSplit.counts, float64(count))
+			currentLength++
 		}
 	}
 
-	return []dataPoint{{
+	//fmt.Println("firstSplit", firstSplit)
+	//fmt.Println("secondSplit", secondSplit)
+
+	var datapoints []dataPoint
+	// Add second data point (last 100 elements or fewer)
```

Comment:
Isn't this adding the first datapoint?

Reply:
The code here was assigning the datapoint and reversely. I have updated to code to add N datapoint sequentially.